component/src/main/java/org/wso2/extension/siddhi/execution/extrema/MaxByMinByTimeWindowProcessor.java

```diff
@@ -248,31 +254,38 @@ public void stop() {
         //Do nothing
     }
 
-    /**
-     * Used to collect the serializable state of the processing element, that need to be
-     * persisted for the reconstructing the element to the same state on a different point of time
-     *
-     * @return stateful objects of the processing element as an array
-     */
     @Override
-    public Map<String, Object> currentState() {
-        return new HashMap<String, Object>() {
-            {
-                put("sortedMap", minByMaxByExecutor.getSortedEventMap());
-            }
-        };
+    public ProcessingMode getProcessingMode() {
+        return ProcessingMode.SLIDE;
     }
 
-    /**
-     * Used to restore serialized state of the processing element, for reconstructing
-     * the element to the same state as if was on a previous point of time.
-     *
-     * @param state the stateful objects of the element as an array on
-     *              the same order provided by currentState().
-     */
-    @Override
-    public void restoreState(Map<String, Object> state) {
-        minByMaxByExecutor.setSortedEventMap((TreeMap) state.get("sortedMap"));
+    static class ExtensionState extends State {
```

Comment:
Any reason to make the class static?

Reply:
These state maintaining classes needed to be static. Otherwise, there will be findbugs related errors thrown.